driver-core/src/main/com/mongodb/operation/CountOperation.java

```diff
@@ -297,4 +354,36 @@ private BsonDocument getCommand(final SessionContext sessionContext) {
         }
         return document;
     }
+
+    private AggregateOperation<BsonDocument> getAggregateOperation() {
+        if (hint != null && !hint.isDocument()) {
```

Comment:
This is supportable by aggregate command:

MongoDB Enterprise repl0:PRIMARY> db.test.getIndexes()
[
	{
		"v" : 2,
		"key" : {
			"_id" : 1
		},
		"name" : "_id_",
		"ns" : "test.test"
	}
]
MongoDB Enterprise repl0:PRIMARY> db.runCommand({aggregate : "test", pipeline : [{$match : {_id : 1}}], cursor : {}, hint : "_id_"})
{
	"cursor" : {
		"firstBatch" : [ ],
		"id" : NumberLong(0),
		"ns" : "test.test"
	},
	"ok" : 1,
	"operationTime" : Timestamp(1529329089, 1),
	"$clusterTime" : {
		"clusterTime" : Timestamp(1529329089, 1),
		"signature" : {
			"hash" : BinData(0,"AAAAAAAAAAAAAAAAAAAAAAAAAAA="),
			"keyId" : NumberLong(0)
		}
	}
}

Reply:
Ack - updated.